compass/landice/tests/greenland/mesh.py

```diff
@@ -56,142 +58,79 @@ def run(self):
         Run this step of the test case
         """
         logger = self.logger
-        mesh_name = 'GIS.nc'
-        section_name = 'mesh'
         config = self.config
-        section = config[section_name]
-        data_path = section.get('data_path')
-        nProcs = section.get('nProcs')
+
+        section_gis = config['greenland']
+
+        nProcs = section_gis.get('nProcs')
+        src_proj = section_gis.get("src_proj")
+        data_path = section_gis.get('data_path')
+        measures_filename = section_gis.get("measures_filename")
+        bedmachine_filename = section_gis.get("bedmachine_filename")
+
+        measures_dataset = os.path.join(data_path, measures_filename)
+        bedmachine_dataset = os.path.join(data_path, bedmachine_filename)
+
+        section_name = 'mesh'
+
+        source_gridded_dataset_1km = 'greenland_1km_2024_01_29.epsg3413.icesheetonly.nc'  # noqa: E501
+        source_gridded_dataset_2km = 'greenland_2km_2024_01_29.epsg3413.nc'
 
         logger.info('calling build_cell_width')
         cell_width, x1, y1, geom_points, geom_edges, floodMask = \
             build_cell_width(
                 self, section_name=section_name,
-                gridded_dataset='greenland_2km_2024_01_29.epsg3413.nc',
+                gridded_dataset=source_gridded_dataset_2km,
                 flood_fill_start=[100, 700])
 
+        # Preprocess the gridded GrIS source datasets to work
+        # with the rest of the workflow
+        logger.info('calling preprocess_gridded_data')
+        preprocessed_gridded_dataset = preprocess_gridded_data(
+            self, source_gridded_dataset_1km, floodMask)
```

Comment:
The array that causes the broadcasting error is half the size `(1408,832)` of the array that's attempting to be broadcast against `(2816,1664)`. My hunch is that this has to do with mixing and matching the 1km and 2km gridded datasets in the highlighted code section. 

That being said in the version of the code on main, there seems to be the same mixing and matching of 1km and 2km datasets. My guess is that this the root of the problem, but why it's a problem with the unified interpolation functions and not the previous version of the code isn't super clear to me yet.